service/controller/resource/appversionlabel/create.go

```diff
@@ -0,0 +1,101 @@
+package appversionlabel
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/giantswarm/cluster-operator/v3/pkg/project"
+	"github.com/giantswarm/cluster-operator/v3/service/controller/key"
+	"github.com/giantswarm/cluster-operator/v3/service/internal/releaseversion"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// The app custom resources are deleted when the namespace is deleted.
+	if key.IsDeleted(&cr) {
+		r.logger.Debugf(ctx, "not updating app version labels for tenant cluster %#q", key.ClusterID(&cr))
+		r.logger.Debugf(ctx, "canceling resource")
+		return nil
+	}
```

Comment:
AFAIK `EnsureCreated` is never executed on delete event. So this block should be fine to remove.

Reply:
Yes true. I got this mixed up with app CRs. 

For app CRs because we delete the namespace we get an update event with the deletion timestamp set. That is not true for cluster CRs.

Done.